p2p/peer/peer.go

```diff
@@ -39,10 +39,11 @@ func (id ID) Loggable() map[string]interface{} {
 func (id ID) String() string {
 	pid := id.Pretty()
 	maxRunes := 6
-	if len(pid) < maxRunes {
-		maxRunes = len(pid)
+	skip := 2 //Added to skip past Qm which is identical for all SHA256 nodes	
+	if len(pid) < maxRunes + skip {
+		maxRunes = len(pid) - skip
 	}
-	return fmt.Sprintf("<peer.ID %s>", pid[:maxRunes])
+	return fmt.Sprintf("<peer.ID %s>", pid[skip:maxRunes + skip])
```

Comment:
I think this will panic if `len(pid) == 0`. 

How about: 

```go
pid := id.Pretty()
if strings.HasPrefix(pid, "Qm") {
  pid = pid[2:]
}
maxRunes := 6
if len(pid) < maxRunes {
  maxRunes = len(pid) 
}
return fmt.Sprintf("<peer.ID %s>", pid[:maxRunes])
```